From 4b29d8ba64e6049561929900ca79b4688e0d057c Mon Sep 17 00:00:00 2001
From: "penikelapati.kumar" <penikelapati.kumar@lge.com>
Date: Wed, 17 Feb 2021 14:12:35 +0530
Subject: [PATCH] Fix crash in p2p/addservice and p2p/settethering

:Release Notes:
FIx crash when p2p/addservice and p2p/settethering.

:Detailed Notes:

:Testing Performed:
Build successfully.
Ethrenet/wifi connection successfully.

:QA Notes:
N/A

:Issues Addressed:
[PLAT-137131] [webOS OSE]: Failed to enable tethering mode
---
 src/group.c | 5 ++---
 src/p2pgo.c | 2 ++
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/group.c b/src/group.c
index 83661cf1..e55358b3 100644
--- a/src/group.c
+++ b/src/group.c
@@ -85,11 +85,9 @@ static int set_tethering(struct connman_group *group,
 			&val);
 
 	if (enabled == TRUE) {
-		__connman_p2p_go_set_enabled();
 		__connman_p2p_go_tethering_set_enabled();
 	} else {
 		__connman_p2p_go_tethering_set_disabled();
-		__connman_p2p_go_set_disabled();
 	}
 
 	return 0;
@@ -554,7 +552,8 @@ static int group_register(struct connman_group *group)
 								group_methods, group_signals,
 								NULL, group, NULL);
 
-	__connman_p2p_set_dhcp_pool(NULL);
+	if (!group->autonomous)
+		__connman_p2p_set_dhcp_pool(NULL);
 	group_added_signal(group);
 
 	return 0;
diff --git a/src/p2pgo.c b/src/p2pgo.c
index e9d5f2d0..5c0b7f79 100644
--- a/src/p2pgo.c
+++ b/src/p2pgo.c
@@ -237,6 +237,8 @@ void __connman_p2p_go_tethering_set_enabled(void)
 	const char *subnet_mask;
 	const char *start_ip;
 
+	if (!dhcp_ippool)
+		return;
 	subnet_mask = __connman_ippool_get_subnet_mask(dhcp_ippool);
 	start_ip = __connman_ippool_get_start_ip(dhcp_ippool);
 
